packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -639,34 +599,39 @@ export class MarketOperationUtils {
             }
         }
 
-        // If RFQ liquidity is enabled, make a request to check RFQ liquidity
+        // Calculate a suggested price. For now, this is simply the overall price of the aggregation.
+        // We can use this as a comparison price for RFQ
         let wholeOrderPrice: BigNumber | undefined;
+        if (optimizerResult) {
+            wholeOrderPrice = getComparisonPrices(
+                optimizerResult.adjustedRate,
+                amount,
+                marketSideLiquidity,
+                _opts.feeSchedule,
+            ).wholeOrder;
+        }
+
+        // If RFQ liquidity is enabled, make a request to check RFQ liquidity against the first optimizer result
         const { rfqt } = _opts;
-        if (rfqt && rfqt.quoteRequestor && marketSideLiquidity.quoteSourceFilters.isAllowed(ERC20BridgeSource.Native)) {
-            // Calculate a suggested price. For now, this is simply the overall price of the aggregation.
-            if (optimizerResult) {
-                wholeOrderPrice = getComparisonPrices(
-                    optimizerResult.adjustedRate,
+        if (
+            marketSideLiquidity.isRfqSupported &&
+            rfqt &&
+            rfqt.quoteRequestor &&
+            marketSideLiquidity.quoteSourceFilters.isAllowed(ERC20BridgeSource.Native)
+        ) {
+            // Timing of RFQT lifecycle
+            const timeStart = new Date().getTime();
+            const { makerToken, takerToken } = nativeOrders[0].order;
+            if (rfqt.isIndicative) {
+                // An indicative quote is being requested, and indicative quotes price-aware enabled
+                // Make the RFQT request and then re-run the sampler if new orders come back.
+                const indicativeQuotes = await rfqt.quoteRequestor.requestRfqtIndicativeQuotesAsync(
+                    makerToken,
+                    takerToken,
                     amount,
-                    marketSideLiquidity,
-                    _opts.feeSchedule,
-                ).wholeOrder;
-            }
-
-            const { isFirmPriceAwareEnabled, isIndicativePriceAwareEnabled } = getPriceAwareRFQRolloutFlags(
-                rfqt.priceAwareRFQFlag,
-            );
```

Comment:
Do all MMs support price-aware queries now? Looks like there's no more checking for flags.

Reply:
Yeah it's now the default for everyone to receive a price to beat